src/client/socksclient.ts

```diff
@@ -441,7 +442,7 @@ class SocksClient extends EventEmitter implements SocksClient {
       this.state = SocksClientState.Error;
 
       // Destroy Socket
-      if (!this._socket.destroyed) {
+      if (this._socket instanceof net.Socket && !this._socket.destroyed) {
```

Comment:
Wouldn't we want to call .destroy() on the stream for instances where _socket is not a net.Socket?

Reply:
@castorw

Reply:
Yeah, I did it kinda bluntly. Fixed now.